### PR TITLE
Use MTR_DEPRECATED in Darwin availability helper.

### PR DESCRIPTION
**Pull request by ksperling-apple**
See project-chip/connectedhomeip#24994

### CODE REVIEW

**Comment by ksperling-apple** [comment]
The Matter build failure due to uncommitted changes is expected, these changes will be committed in the Matter repo alongside adopting the next zap version.

**Review by bzbarsky-apple — APPROVED** [review]

File: src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js [hunk 1]
```diff
@@ -734,7 +734,7 @@ async function availability(clusterName, options) {
   let availabilityStrings = Object.entries(introducedVersions).map(
     ([os, version]) => `${os}(${version}, ${deprecatedVersions[os]})`
   );
-  return `API_DEPRECATED("${
```
Comment (brdandu) [inline src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js:737]: Do we need to worry about backward's compatibility here?

Comment (bzbarsky-apple) [reply src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js:737]: In practice, not really.  I don't think anyone will be using new ZAP to regenerate the Darwin APIs on top of an old SDK revision....

Comment (bzbarsky-apple) [reply src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js:737]: Yes, having the compatibility check only check app regen would make a lot of sense.

**Pull request merged**